packages/validator/lib/validatorHelper.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+function getLastParamName(paramPath) {
+  const lastDotIndex = paramPath.lastIndexOf('.');
+  let paramName = paramPath;
+
+  if (lastDotIndex >= 0) {
+    paramName = paramPath.slice(lastDotIndex + 1);
+  }
+
+  return paramName;
+}
+
+function replaceConditionalMatch(match, specs) {
+  let parsedSpecs = {};
+
+  for (const key of Object.keys(specs)) {
+    if (key === '__conditions') {
+      continue;
+    }
+
+    let newKey = key.replace(/__match/g, match);
+    parsedSpecs[newKey] = typeof specs[key] === 'string' ? specs[key].replace(/__match/g, match) : replaceConditionalMatch(match, specs[key]);
+  }
+
+  return parsedSpecs;
+}
+
+function checkRedundancy(nonRedundant, specs, paramPath, messages) {
+  if (typeof specs === 'object') {
```

Comment:
Updated in 85be84a
In case the specs is array, regular `for` loop seems like the best solution. If `map` would be used the result needs flattened, which can be avoided by using `for`...